### ISSUE TITLE
[BUG] MobileWebAuth login doesn't work

### ISSUE THREAD

**Issue by voviz** [ValvePython/steam#442]
**Description**
MobileWebAuth login doesn't work. Function _finalise_login throws an error - 
```
line 323, in _finalize_login
    data = json.loads(login_response['oauth'])
                      ~~~~~~~~~~~~~~^^^^^^^^^
KeyError: 'oauth'
```

Everything worked fine until today, so probably the response body changed. I have changed the function to
 ```
 def _finalize_login(self, login_response):
        data = login_response['transfer_parameters']
        self.steam_id = SteamID(data['steamid'])
        self.oauth_token = data['auth']
```
And the login seems to be ok. However steam guard can not be added
Code - 
```
wa = MobileWebAuth("LOGIN")
wa.cli_login(password="PASSWORD")
sa = SteamAuthenticator(backend=wa)
status = sa.status()
```
Exception - 
```
 line 267, in status
    return self._send_request('QueryStatus', {'steamid': self.backend.steam_id})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  line 155, in _send_request
    raise SteamAuthenticatorError("Error adding via WebAPI: %s" % str(exp))
steam.guard.SteamAuthenticatorError: Error adding via WebAPI: 401 Client Error: Unauthorized for url: https://api.steampowered.com/ITwoFactorService/QueryStatus/v1/
```

**Comment by qvchoq** [comment]
Поплачь блядота

**Comment by artemiyDev** [comment]
also have this problem. I think steam have changed auth process

**Comment by arynyklas** [comment]
Same problem

**Comment by Zachxz0** [comment]
@voviz had you solved it ?

**Comment by serveroid** [comment]
+1

**Comment by serveroid** [comment]
> **Description** MobileWebAuth login doesn't work. Function _finalise_login throws an error -
> 
> ```
> line 323, in _finalize_login
>     data = json.loads(login_response['oauth'])
>                       ~~~~~~~~~~~~~~^^^^^^^^^
> KeyError: 'oauth'
> ```
> 
> Everything worked fine until today, so probably the response body changed. I have changed the function to
> 
> ```
> def _finalize_login(self, login_response):
>        data = login_response['transfer_parameters']
>        self.steam_id = SteamID(data['steamid'])
>        self.oauth_token = data['auth']
> ```
> 
> And the login seems to be ok. However steam guard can not be added Code -
> 
> ```
> wa = MobileWebAuth("LOGIN")
> wa.cli_login(password="PASSWORD")
> sa = SteamAuthenticator(backend=wa)
> status = sa.status()
> ```
> 
> Exception -
> 
> ```
>  line 267, in status
>     return self._send_request('QueryStatus', {'steamid': self.backend.steam_id})
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   line 155, in _send_request
>     raise SteamAuthenticatorError("Error adding via WebAPI: %s" % str(exp))
> steam.guard.SteamAuthenticatorError: Error adding via WebAPI: 401 Client Error: Unauthorized for url: https://api.steampowered.com/ITwoFactorService/QueryStatus/v1/
> ```

find any solution?

**Comment by Rosuav** [comment]
> I have changed the function to
> 
> ```
> def _finalize_login(self, login_response):
>        data = login_response['transfer_parameters']
>        self.steam_id = SteamID(data['steamid'])
>        self.oauth_token = data['auth']
> ```
> 
> And the login seems to be ok. However steam guard can not be added

Are you able to get trades to work? I haven't been able to.

**Comment by REDNBLACK** [comment]
Library probably needs to implement new OAuth based login and SteamGuard management to work, like here: https://github.com/geel9/SteamAuth/commit/d44e99dc976cb5a21185e1e250d46235f20f6968

**Comment by hani9898** [comment]
having the same problem

**Comment by h3902340** [comment]
same issue. Need to fix asap.

**Comment by asa** [comment]
Also hitting this.

**Comment by awxk** [comment]
This is still a big problem.

**Comment by nan4k7** [comment]
Same problem here... any support?

**Comment by lkurgan55** [comment]
Find any solution? Help pls <3

**Comment by lkurgan55** [comment]
Maybe there is another way to register guard?

**Comment by masterwishx** [comment]
is issue fixed ?

**Comment by Heehes** [comment]
Is anyone working on fixing this?

**Comment by lkurgan55** [comment]
Nope, use SDA, the best way to register and use steam authenticator

**Comment by Rosuav** [comment]
Elaborate?

**Comment by Heehes** [comment]
Is there any other guard generated python library

**Comment by zahrasalehi** [comment]
any update?

**Comment by corytheboyd** [comment]
One+ years later, and I am running into the same issue. Guessing Valve abandoned this project, but didn't archive it :/

**Comment by teohhanhui** [comment]
@corytheboyd

> This organization is not affiliated with Valve Software or Steam.

**Comment by corytheboyd** [comment]
> [@corytheboyd](https://github.com/corytheboyd)
> 
> > This organization is not affiliated with Valve Software or Steam.

Oh, I see.
